docs/src/p4est_interfacing.md

```diff
@@ -0,0 +1,333 @@
+# Interfacing to the `p4est` mesh backend
+
+Sometimes it is necessary to send data from the solver to the mesh backend.
+In the case of [`p4est`](https://www.p4est.org/) these are relatively decoupled, in contrast to the other mesh types.
```

Comment:
```suggestion
Sometimes it is necessary to exchange data between the solver and the mesh.
In the case of the [`P4estMesh`](@ref) and its [`p4est`](https://www.p4est.org/), this is more intricate than with other mesh types, since the p4est backend
is comparatively opaque to the rest of the code due to being a compiled C library.
```